lib/cpp/src/thrift/concurrency/ThreadManager.cpp

```diff
@@ -421,7 +416,7 @@ void ThreadManager::Impl::removeWorker(size_t value) {
   {
     Synchronized s(workerMonitor_);
 
-    while (workerCount_ != workerMaxCount_) {
+    while (workerCount_ > goalCount) {
```

Comment:
In that particular scenario, the second removeWorker should fail since we should be synchronizing correctly when it checks if (value > workerMaxCount_), but I agree that ThreadManager probably needs some TLC.